sentence_transformers/SentenceTransformer.py

```diff
@@ -1332,6 +1354,28 @@ def _load_auto_model(
         self.model_card_data.set_base_model(model_name_or_path, revision=revision)
         return [transformer_model, pooling_model]
 
+    def _load_module_class_from_ref(
+        self, class_ref: str, model_name_or_path: str, trust_remote_code: bool, model_kwargs: Optional[Dict[str, Any]]
+    ) -> nn.Module:
+        # If the class is from sentence_transformers, we can directly import it,
+        # otherwise, we try to import it dynamically, and if that fails, we fall back to the default import
+        if class_ref.startswith("sentence_transformers."):
+            return import_from_string(class_ref)
+
+        if trust_remote_code:
+            code_revision = model_kwargs.pop("code_revision", None) if model_kwargs else None
+            try:
+                return get_class_from_dynamic_module(
+                    class_ref,
+                    model_name_or_path,
+                    code_revision=code_revision,
+                )
+            except EnvironmentError:
+                # Ignore the error if the file does not exist, and fall back to the default import
+                pass
+
+        return import_from_string(class_ref)
```

Comment:
@muellerzr I'd love to get your opinion on whether this might be problematic in terms of security. The first `if` imports something from the Sentence Transformers module, which should be fine unless a malicious package is installed that uses `sentence_transformers` as the import name.

The second `if` imports the class via the `transformers` remote code functionality. I assume this is safe.

The third `if` imports whatever the class reference is, so this can also be a class in a malicious module. But the user would already have to have that malicious package installed. I'm not too concerned about this case, as preventing this case also prevents third parties from creating non-malicious custom ST Modules. 

- Tom Aarsen